blocksync/blocksync_v2.go

```diff
@@ -0,0 +1,290 @@
+package blocksync
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/iotexproject/iotex-core/blockchain/block"
+	"github.com/iotexproject/iotex-core/pkg/fastrand"
+	"github.com/iotexproject/iotex-core/pkg/lifecycle"
+	"github.com/iotexproject/iotex-core/pkg/log"
+	"github.com/iotexproject/iotex-core/pkg/routine"
+	"github.com/iotexproject/iotex-proto/golang/iotexrpc"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+type blockSyncerV2 struct {
```

Comment:
looks like the struct and funcs are mostly same as current code, we don't need a V2? can just change on current code

Reply:
yes, added V2 is to compare it with the current one.